apis/concierge/identity/types_whoami.go.tmpl

```diff
@@ -0,0 +1,40 @@
+// Copyright 2021 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package identity
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WhoAmIRequest submits a request to echo back the current authenticated user.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type WhoAmIRequest struct {
```

Comment:
Minor naming nit, but I might also consider something like `SelfIdentityRequest` or `SelfSubjectIdentityReview` just for stylistic consistency. I'm fine with this too, though.

Reply:
😬 auto merge was set so this merged before we had a chance to discuss this.

Discussed post merge on slack.  My thought process had been "what API would the future `pinniped whoami` command call under the hood?"  `SelfSubjectIdentityReview` is not bad because it matches the existing Kube API `SelfSubjectRulesReview`.  Not sure if either of `WhoAmIRequest` or `SelfSubjectRulesReview` would be clear to an end user.  Maybe it does not really matter because end users would just use the CLI in the future anyway?